scalr/resource_scalr_variable.go

```diff
@@ -34,16 +34,26 @@ func resourceScalrVariable() *schema.Resource {
 				// Reject any changes for variable scope
 				var scopeAttributes = []string{"workspace_id", "environment_id", "account_id"}
 
-				notChangedFields := 0
+				scopeIsAlreadySet := false
```

Comment:
Why not just to use:
```golang

func(d *schema.ResourceDiff, meta interface{}) error {
	// Reject any changes for variable scope
	if d.HasChange("workspace_id") || d.HasChange("environment_id") || d.HasChange("account_id") {
		return fmt.Errorf("Error changing scope for variable %s: scope is immutable attribute", d.Id())
	}
	return nil
},
```

Reply:
Good point, but if variable is about to create, all of these parameters has change, and our `CustomizeDiff` raise an error. That's why we need to check if scope is already set previously (`environment_id`, `workspace_id` or `account_id` are not empty)